src/rgw/rgw_sal_motr.cc

```diff
@@ -2137,6 +2137,7 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
       source->gen_rand_obj_instance_name();
       std::string del_marker_ver_id = source->get_instance();
       result.version_id = del_marker_ver_id;
+      source->delete_marker = true;
```

Comment:
at line 2136 also we are setting delete_marker on result object. 
Please check if we need to set this boolean value for both result & source object.

Reply:
Yes
`Results` structure contains delete-marker field to identify that delete-marker entry is present for an object and should be visible when we perform list-object-versions.
Here, we are setting the delete-marker to true to visible the response fields such as "DeleteMarker, DeleteMarkerVersionId" in the DeleteObjects API response.